modules/codegen/src/main/scala/com/twilio/guardrail/generators/Java/JacksonGenerator.scala

```diff
@@ -0,0 +1,580 @@
+            new VariableDeclarationExpr(new VariableDeclarator(enumType, "value"), FINAL),
+            new MethodCallExpr("values"),
+            new BlockStmt(new NodeList(
+              new IfStmt(
+                new MethodCallExpr("value.name.equals", new NameExpr("name")),
+                new ReturnStmt(new NameExpr("value")),
+                null
+              )
+            ))
+          ),
+          new ThrowStmt(new ObjectCreationExpr(
+            null,
+            JavaParser.parseClassOrInterfaceType("IllegalArgumentException"),
+            new NodeList(
+              new BinaryExpr(
+                new BinaryExpr(new StringLiteralExpr("Name '"), new NameExpr("name"), BinaryExpr.Operator.PLUS),
+                new StringLiteralExpr(s"' is not valid for enum '${clsName}'"),
+                BinaryExpr.Operator.PLUS
+              )
+            )
+          ))
+        )))
+
+
+        val enumClass = new EnumDeclaration(
+          util.EnumSet.of(PUBLIC),
+          new NodeList(),
+          new SimpleName(clsName),
+          new NodeList(),
+          new NodeList(enumDefns: _*),
+          new NodeList(
+            nameField,
+            constructor,
+            getNameMethod,
+            parseMethod
+          )
+        )
+
+        Target.pure(enumClass)
+        //Target.raiseError(enumClass.toString)
+
+      case RenderStaticDefns(clsName, members, accessors, encoder, decoder) =>
+        for {
+          extraImports <- List(
+            "com.fasterxml.jackson.annotations.JsonCreator",
+            "com.fasterxml.jackson.annotations.JsonValue"
+          ).map(safeParseRawImport).sequence
+        } yield StaticDefns[JavaLanguage](
+          className = clsName,
+          extraImports = extraImports,
+          definitions = List.empty
+        )
+
+      case BuildAccessor(clsName, termName) =>
+        Target.pure(new Name(s"${clsName}.${termName}"))
+    }
+  }
+
+  object ModelProtocolTermInterp extends (ModelProtocolTerm[JavaLanguage, ?] ~> Target) {
+    def apply[T](term: ModelProtocolTerm[JavaLanguage, T]): Target[T] = term match {
+      case ExtractProperties(swagger) =>
+        (swagger match {
+          case m: ObjectSchema => Target.pure(Option(m.getProperties))
+          case comp: ComposedSchema =>
+            Target.pure(Option(comp.getAllOf()).toList.flatMap(_.asScala.toList).lastOption.flatMap(prop => Option(prop.getProperties)))
+          case comp: Schema[_] if Option(comp.get$ref).isDefined =>
+            Target.raiseError(s"Attempted to extractProperties for a ${comp.getClass()}, unsure what to do here")
+          case _ => Target.pure(None)
+        }).map(_.map(_.asScala.toList).toList.flatten)
+
+      case TransformProperty(clsName, name, property, meta, needCamelSnakeConversion, concreteTypes, isRequired) =>
+        for {
+          _ <- Target.log.debug("definitions", "circe", "modelProtocolTerm")(s"Generated ProtocolParameter(${term}, ${name}, ...)")
```

Comment:
The new form of this is 
```scala
Target.log.function("transformProperty($clsName, $name, $property, $meta, $needsCamelSnakeConversion, $concreteTypes, $isRequired)") {
  for {
    _ <- Target.log.debug(s"Generated ProtocolParameter($term, $name, ...)")
```
This is stack-based, using `StateT` to track where in the logging stack we are, so calling the same method with different paths will give you the call path to get there in the debug logs

Reply:
Sorry, not sure what you're asking me to change -- I think this was c/p from circe when you first created the stub...?

Reply:
(Well, aside from fixing up that it says circe and should say jackson in the log message)

Reply:
Yeah -- my initial comment was `circe` related, but when I did #188, I haven't gone through to wire in the new logging infrastructure (part of why there are so many warnings everywhere now). I'm only suggesting that you can use the new logging  infrastructure to slightly reduce warnings on new code being added.